smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -104,15 +106,42 @@ contract MixinPurchase is
     }
   }
 
+  /** 
+   * @param _baseAmount the total amount to calculate the fee
+   * @dev internal function to execute the payments to referrers if any is set
+   */
+  function _payProtocol(uint _baseAmount) internal {
+
+    // get fee from Unlock 
+    uint protocolFee;
+    try unlockProtocol.protocolFee() {
+      // calculate fee to be paid
+      protocolFee = (_baseAmount * unlockProtocol.protocolFee()) / BASIS_POINTS_DEN;
```

Comment:
Is this making 2 calls to `unlock` contract? Maybe better to use a variable so we avoid spending gas?